framework/src/main/java/org/checkerframework/common/value/util/Range.java

```diff
@@ -40,65 +42,81 @@
 
     /** A range containing all possible 8-bit values. */
-    public static final Range BYTE_EVERYTHING = new Range(Byte.MIN_VALUE, Byte.MAX_VALUE);
+    public static final Range BYTE_EVERYTHING = create(Byte.MIN_VALUE, Byte.MAX_VALUE);
 
-    /** The empty range. */
-    public static final Range NOTHING = new Range();
+    /** The empty range singleton. */
+    public static final Range NOTHING = new Range(Long.MAX_VALUE, Long.MIN_VALUE);
 
     /** An alias to the range containing all possible 64-bit values. */
     public static final Range EVERYTHING = LONG_EVERYTHING;
 
     /**
-     * Create a Range from a collection of Longs.
+     * Constructs a range with its bounds specified by two parameters, {@code from} and {@code to}.
      *
-     * @param collection from which min and max values will be used as from and to values in the
-     *     range
-     */
-    public Range(Collection<Long> collection) {
-        this(Collections.min(collection), Collections.max(collection));
-    }
-
-    /**
-     * Create a Range from a collection of Integers.
+     * <p>Private constructor does no validation of arguments so special instances can be created
+     * through it.
+     *
+     * <p>Example {@code #NOTHING}.
      *
-     * @param list from which min and max values will be used as from and to values in the range
+     * @param from the lower bound (inclusive)
+     * @param to the upper bound (inclusive)
      */
-    // The formal parameter type cannot be Collection<Integer> because that would erase to the same
-    // raw types as the Range(Collection<Long>) constructor.
-    public Range(List<Integer> list) {
-        this(Collections.min(list), Collections.max(list));
+    private Range(long from, long to) {
+        this.from = from;
+        this.to = to;
     }
 
     /**
      * Constructs a range with its bounds specified by two parameters, {@code from} and {@code to}.
      *
+     * <p>Public factory method validates that {@code from <= to}
+     *
      * @param from the lower bound (inclusive)
      * @param to the upper bound (inclusive)
+     * @return a valid Range for {@code from <= to}, otherwise IllegalArgumentException.
      */
-    public Range(long from, long to) {
+    public static Range create(long from, long to) {
         if (!(from <= to)) {
             throw new IllegalArgumentException(String.format("Invalid Range: %s %s", from, to));
         }
-        this.from = from;
-        this.to = to;
+        return new Range(from, to);
     }
 
-    /** Creates the singleton empty range. */
-    private Range() {
-        this.from = Long.MAX_VALUE;
-        this.to = Long.MIN_VALUE;
+    /**
+     * Create a Range from a collection of Number, Longs, Integer.
+     *
+     * <p>This is a public convenience factory method that builds using the public factory for a
+     * range.
+     *
+     * @param values never null collection from which min and max values will be used as from and to
```

Comment:
We usually use the non-null default that the Nullness Checker uses. So if you don't say that null is allowed, it's forbidden. If something does allow null, it should be annotated as `@Nullable` and explain in the javadoc.

Reply:
Thanks. I was not sure if this code base itself was subjected to the Nullness Checker .  
I will remove the runtime check on a commit to come.

Reply:
We're not running the Nullness Checker yet (#2916 is starting work on that), but we already follow that style.